packages/types/src/interfaces/extrinsic/v1/Extrinsic.ts

```diff
@@ -79,4 +82,28 @@ export default class CENNZnetExtrinsic extends GenericExtrinsic{
       })
     return this;
   }
+
+  private cvmToAddress(cvmAddress) {
+    let message = stringToU8a('cvm:');
+    message = u8aConcat(message, new Array(7).fill(0), hexToU8a(cvmAddress));
+    const checkSum = message.reduce((a, b) => a ^ b, 0);
+    message = u8aConcat(message, new Array(1).fill(checkSum));
+
+    return encodeAddress(message, 42);
+  }
+
+  /**
+   * @description sign the extrinsic via metamask
+   */
+  async signViaMetaMask(ethAddress: string, api: Api): Promise<SubmittableResultResult<ApiTypes>> {
```

Comment:
name shouldn't be 'metamask'
other ethereum wallets can use this too: https://docs.walletconnect.com/json-rpc-api-methods/ethereum